Eliminate c10/util/array from PyTorch

r-barnes opened the pull request:
Test Plan: Sandcastle

Differential Revision: D46772319

Skylion007 reviewed (approved):
LGTM assuming all tests pass and this is really a drop-in replacement.

Skylion007 commented on c10/util/Array.h:
```diff
@@ -1,391 +1,13 @@
-} // namespace detail
-
-template <class T, size_t N>
-constexpr array<T, N> to_array(const T (&arr)[N]) {
```
Do we not need to alias c10:to_array as well? Was this path unused?

r-barnes committed:
Eliminate c10/util/array from PyTorch (#103893)

Summary: Pull Request resolved: https://github.com/pytorch/pytorch/pull/103893

Test Plan: Sandcastle

Differential Revision: D46772319

fbshipit-source-id: 3b3082ffd217a83e325e1f80865abbd3a638d779
